Tests/Tests/CommonUtilitySpec.swift

```diff
@@ -104,6 +104,38 @@ class CommonUtilitySpec: QuickSpec {
                 }
             }
 
+            context("CommonUtility.convertTriggerObjectToCustomEvent") {
+
+                it("will return Custom Event") {
+                    let attribute = TriggerAttribute(name: "att", value: "val",
+                                                     type: .string, operatorType: .isNotBlank)
+                    let trigger = Trigger(type: .event, eventType: .appStart, eventName: "app Start", attributes: [attribute])
+                    let result = CommonUtility.convertTriggerObjectToCustomEvent(trigger)
+                    expect(result).toNot(beNil())
+                }
+            }
+
+            context("CommonUtility.unlockResource") {
+
+                it("will lock resource object") {
+                    let lockableObject = LockableTestObject()
+                    CommonUtility.unlock(resourcesIn: lockableObject)
+                    expect(lockableObject.resource.get()).toEventually(equal([]))
+                }
+            }
+
+            context("CommonUtility.lockResource") {
+
+                it("will unlock resource object") {
+                    let lockableObject = LockableTestObject()
+                    CommonUtility.lock(resourcesIn: lockableObject)
+                    DispatchQueue.global().asyncAfter(deadline: .now() + 1, execute: {
+                        lockableObject.append(1)
```

Comment:
this might be executed after `expect(lockableObject.resource.get()).toEventually(equal([]))`. There's a race condition here. In this scenario I think it's best to use `waitUntil`:
```
waitUntil { done in
 DispatchQueue.global().asyncAfter(deadline: .now() + .milliseconds(500), execute: {
     lockableObject.append(1)
     done()
 }
}
expect(lockableObject.resource.get()).toAfterTimeout(equal([]))
```

Reply:
And just in case I would unlock `lockableObject` after the test is completed. Ideally in `afterEach`